src/main/java/org/apache/commons/io/IOUtils.java

```diff
@@ -2697,6 +2724,62 @@ public static byte[] toByteArray(final InputStream input, final long size) throw
         return toByteArray(input, (int) size);
     }
 
+    /**
+     * Reads exactly {@code size} bytes from the given {@link InputStream} into a new {@code byte[]}.
+     *
+     * <p>
+     *   This variant validates that the stream actually contains {@code size} bytes.
+     *   It is suitable for untrusted input because it prevents oversized allocations when the provided {@code size}
```

Comment:
I don't think we should start to document which APIs are designed for trusted versus untrusted input. The documentation should just describe the intended functionality.